x/launch/keeper/msg_server_create_chain.go

```diff
@@ -27,8 +27,8 @@ func (k msgServer) CreateChain(goCtx context.Context, msg *types.MsgCreateChain)
 		msg.SourceHash,
 		msg.GenesisURL,
 		msg.GenesisHash,
-		false,
```

Comment:
The call of `CreateNewChain` had previously all the preconditions so that it is supposed to never fail. Hence, the fact we were handling the error with a critical error.

However, now the method can fail because of a non-existing campaign.

Since a critical error should not be able to be triggered by a user, we should either handle this method with a normal error or add a check before to check the campaign exist and let the critical error here (because the preconditions would then be met again)

I'm in favor of the 2nd personally

Reply:
Also, the method can fail because the coordinator of the chain and the campaign don't match. So maybe it can be simpler to just directly handle this method with an normal error